crates/evm/evm/src/inspectors/stack.rs

```diff
@@ -386,17 +396,25 @@ impl InspectorStack {
 
     /// Set whether to enable the tracer.
     #[inline]
-    pub fn tracing(&mut self, yes: bool) {
-        self.tracer = yes.then(|| {
-            TracingInspector::new(TracingInspectorConfig {
-                record_steps: false,
-                record_memory_snapshots: false,
-                record_stack_snapshots: StackSnapshotType::None,
-                record_state_diff: false,
-                exclude_precompile_calls: false,
-                record_logs: true,
-            })
-        });
+    pub fn tracing(&mut self, yes: bool, debug: bool) {
+        if self.tracer.is_none() && yes ||
+            !self.tracer.as_ref().map_or(false, |t| t.config().record_steps) && debug
```

Comment:
this is a bit hard to follow,
I wonder if we can encapsulate these two bools into an enum `TracingKid` or smth, because debug also implies tracing, right?